Assets/Scripts/CarController.cs

```diff
@@ -99,6 +100,8 @@ private void Update()
     {
         UpdateWheelPoses();
         ProcessBotCommands();
+        var velocityAngle = Vector3.Angle(rigidBody.transform.forward, rigidBody.velocity);
+        velocity = rigidBody.velocity.magnitude * ((velocityAngle <= 90) ? 1 : -1);
```

Comment:
Maybe this could be :
velocity = Vector3.Dot(rigidBody.transform.forward, rigidBody.velocity)

Reply:
Done. Had totally forgot my vector math...